File: Unique In Order/unique-in-order.py
def unique_in_order(iterable):
    """
    :type iterable: str|list
    :rtype: list
    """
    res = []

    if len(iterable):
        item = iterable[0]
        res.append(item)
        for it in iterable[1:]:
            if it != item:
                item = it
                res.append(item)

    return res


print(unique_in_order('AAAABBBCCDAABBB'))  # ['A', 'B', 'C', 'D', 'A', 'B']
print(unique_in_order('ABBCcAD'))  # ['A', 'B', 'C', 'c', 'A', 'D']
print(unique_in_order([1, 2, 2, 3, 3]))  # [1, 2, 3]

File: Unique In Order/unique-in-order.php
<?php
/**
 * Created by PhpStorm.
 * User: kovalevcon
 * Date: 03.10.18
 * Time: 17:13
 */

/**
 * Unique In Order function
 *
 * @param string|array $iterable
 * @return array
 */
function uniqueInOrder($iterable): array
{
    $len = is_array($iterable) ? count($iterable) : strlen($iterable);
    $res = [];

    if ($len) {
        $item = $iterable[0];
        $res[] = $item;

        for($i = 0; $i < $len; $i++) {
            if ($iterable[$i] !== $item) {
                $item = $iterable[$i];
                $res[] = $item;
            }
        }
    }

    return $res;
}

echo "<pre>uniqueInOrder(\"AAAABBBCCDAABBB\")\r\n";
var_dump(uniqueInOrder("AAAABBBCCDAABBB")); // ['A', 'B', 'C', 'D', 'A', 'B']
echo "</pre>";

echo "<pre>uniqueInOrder(\"ABBCcAD\")\r\n";
var_dump(uniqueInOrder("ABBCcAD")); // ['A', 'B', 'C', 'c', 'A', 'D']
echo "</pre>";

echo "<pre>uniqueInOrder([1,2,2,3,3])\r\n";
var_dump(uniqueInOrder([1,2,2,3,3])); // [1, 2, 3]
echo "</pre>";